core/domain/question_services.py

```diff
@@ -304,3 +325,30 @@ def get_questions_batch(
 
     questions_batch = get_questions_by_ids(random_question_ids)
     return questions_batch
+
+
+def get_question_summaries_for_collection(collection_id):
+    """Gets a list of question summaries for a collection.
+
+    Args:
+        collection_id: str. ID of the collection.
+
+    Returns:
+        list(QuestionSummary). A list of Question Summary objects.
+    """
+    collection = collection_services.get_collection_by_id(collection_id)
+    skill_ids_of_question = {}
+    for skill in collection.skills.values():
+        for question_id in skill.question_ids:
+            if question_id not in skill_ids_of_question:
+                skill_ids_of_question[question_id] = []
+            skill_ids_of_question[question_id].append(skill.name)
```

Comment:
skill.name is not the same as a skill ID. Be careful with the naming and make sure you're adding the correct thing to the dict.

Reply:
It is skill name. Updated.